src/module_PI/Raspberry_PI/DataPacket.java

```diff
@@ -0,0 +1,136 @@
+package module_PI.Raspberry_PI;
+
+import java.security.MessageDigest;
+import java.security.NoSuchAlgorithmException;
+
+public class DataPacket {
+
+	private final int SIZE_OF_SHA1_IN_BYTES = 8;
+	private final int SIZE_OF_DATAPACKET_IN_BYTES = 64;
+	private int arrayCounter = 1;
+	private byte[] sha1MessageHash;
+	private byte[] packetAsArray;
+	private String messageString;
+	private String packetTypeString;
+	protected MessageDigest digest;
+
+	public DataPacket(String message, String packetType) {
+		this.messageString = message.toUpperCase();
+		this.packetTypeString = packetType.toUpperCase();
+		sha1MessageHash = new byte[SIZE_OF_SHA1_IN_BYTES];
+		packetAsArray = new byte[SIZE_OF_DATAPACKET_IN_BYTES];
+		packetAsArray[0] = '~';
+		initializeMessageDigestForSha1();
+		appendPacketHeader();
+		appendMessage();
+	}
+
+	public DataPacket(byte[] packetContents) {
+		sha1MessageHash = new byte[SIZE_OF_SHA1_IN_BYTES];
+		messageString = "";
+		initializeMessageDigestForSha1();
+		this.packetAsArray = packetContents;
+		parseSha1FromRawPacket(packetContents);
+		parsePacketTypeFromRawPacket(packetContents);
+		parseMessageFromRawPacket(packetContents);
+	}
+
+	public byte[] getPacketAsArray() {
+		return packetAsArray;
+	}
+
+	public byte[] getPacketSha1Hash() {
+		return sha1MessageHash;
+	}
+
+	public int getSizeInBytes() {
+		return SIZE_OF_DATAPACKET_IN_BYTES;
+	}
+	
+	String getPacketType(){
+		  return packetTypeString;
+	}
+
+	String getMessage(){
+	  return messageString;
+	}
+
+	private void appendPacketHeader() {
+		appendHash();
+		appendPacketTypeByte();
+	}
+
+	private void initializeMessageDigestForSha1() {
+		try {
+			digest = MessageDigest.getInstance("SHA-1");
+		} catch (NoSuchAlgorithmException e) {
+			e.printStackTrace();
+		}
+	}
+
+	private void appendHash() {
+		byte[] sha1Hash = digest.digest(messageString.getBytes());
+		for (int i = 0; i < SIZE_OF_SHA1_IN_BYTES; i++) {
+			this.sha1MessageHash[i] = sha1Hash[i];
+		}
+		append(sha1MessageHash);
+	}
+
+	private void appendPacketTypeByte() {
+		byte packetTypeByte = 0;
+		if (packetTypeString == "LOG") {
```

Comment:
Same comment as in the .cpp version of these class. These should, at the very least, be pulled into constants so that it limits the possibility of a bug creeping into this class in the future.

Since enums are trivial to make in Java I'd recommend you do that approach.